### ISSUE TITLE
BadRequest EventHub Metrics

### ISSUE THREAD

**Issue by IMAMBAKS** [tomkerkhove/promitor#2023]
Currently I'm retrieving two metrices: 

 - Incoming messages
 - EventHub size.
 
The `Incoming Messages`  do work however I'm getting BadRequests for other Metrices e.g. `EventHub Size`.

Using the following configuration:
```
metrics:
  - name: azure_event_hubs_incoming_messages
    description: "The number of incoming messages on an Azure Event Hubs topic"
    resourceType: EventHubs
    azureMetricConfiguration:
      metricName: IncomingMessages
      aggregation:
        type: Total
    resources:
      - namespace: xxx-dl-evh-acc

  - name: azure_event_hubs_size
    description: "Size of an EventHub in Bytes."
    resourceType: EventHubs
    azureMetricConfiguration:
      metricName: Size
      aggregation:
        type: Average
    resources:
      - namespace: xxx-dl-evh-acc

metricSinks:
  prometheusScrapingEndpoint:
    enabled: true
    baseUriPath: /metrics
    enableMetricTimestamps: true
    metricUnavailableValue: NaN
    labelTransformation: None
    enableServiceDiscovery: true
    serviceMonitor:
      enabled: true
      namespace: "datalake-monitoring"
      labels: { }
      interval: 30s
      timeout: 10s

metricDefaults:
  aggregation:
    interval: 00:01:00
  scraping:
    schedule: "* * * * *"
```
Bad request message:
```BadRequest: Failed to serialize response for request: https://westeurope.mdmrp.prod.windows.azure.net/subscriptions/xxxxb/resourceGroups/eventhub-acc-rg/providers/Microsoft.EventHub/namespaces/xxx-dl-evh-acc/providers/microsoft.insights/metrics?$filter=EntityName eq '*'&timespan=2022-04-28T07%3A39%3A00.9960246Z%2F2022-05-03T07%3A39%3A00.9960246Z&interval=PT1M&metricnames=Size&aggregation=Average&top=10000&api-version=2018-01-01, Duration: 00:00:00.7031398, TraceId: {4f634207-2197-4058-92c0-ca5932a17b9f}```

Error message:
```
[07:40:02 FTL] Failed to scrape resource for metric 'azure_event_hubs_size'
Microsoft.Azure.Management.Monitor.Fluent.Models.ErrorResponseException: Operation returned an invalid status code 'BadRequest'
   at Microsoft.Azure.Management.Monitor.Fluent.MetricsOperations.ListWithHttpMessagesAsync(String resourceUri, ODataQuery`1 odataQuery, String timespan, Nullable`1 interval, String metricnames, String aggregation, Nullable`1 top, String orderby, Nullable`1 resultType, String metricnamespace, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Monitor.Fluent.MetricsOperationsExtensions.ListAsync(IMetricsOperations operations, String resourceUri, ODataQuery`1 odataQuery, String timespan, Nullable`1 interval, String metricnames, String aggregation, Nullable`1 top, String orderby, Nullable`1 resultType, String metricnamespace, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Monitor.Fluent.Models.MetricDefinitionImpl.ExecuteAsync(CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Monitor.Fluent.Models.MetricDefinitionImpl.Microsoft.Azure.Management.Monitor.Fluent.IWithMetricsQueryExecute.ExecuteAsync(CancellationToken cancellationToken)
   at Promitor.Integrations.AzureMonitor.AzureMonitorClient.GetRelevantMetric(String metricName, AggregationType metricAggregation, TimeSpan metricInterval, String metricFilter, String metricDimension, IMetricDefinition metricDefinition, Nullable`1 metricLimit, DateTime recordDateTime) in /src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs:line 183
   at Promitor.Integrations.AzureMonitor.AzureMonitorClient.QueryMetricAsync(String metricName, String metricDimension, AggregationType aggregationType, TimeSpan aggregationInterval, String resourceId, String metricFilter, Nullable`1 metricLimit) in /src/Promitor.Integrations.AzureMonitor/AzureMonitorClient.cs:line 90
   at Promitor.Core.Scraping.AzureMonitorScraper`1.ScrapeResourceAsync(String subscriptionId, ScrapeDefinition`1 scrapeDefinition, TResourceDefinition resourceDefinition, AggregationType aggregationType, TimeSpan aggregationInterval) in /src/Promitor.Core.Scraping/AzureMonitorScraper.cs:line 72
   at Promitor.Core.Scraping.Scraper`1.ScrapeAsync(ScrapeDefinition`1 scrapeDefinition) in /src/Promitor.Core.Scraping/Scraper.cs:line 103
```

**Comment by tomkerkhove** [comment]
Odd, I will have to check that.

Would you mind enabling this to see if you get more feedback?

https://docs.promitor.io/v2.6/scraping/runtime-configuration/?h=monitor#azure-monitor

**Comment by IMAMBAKS** [comment]
I added the following to my config:

```
azureMonitor:
  logging:
    informationLevel: Basic # Optional. Default: Basic
    isEnabled: true # Optional. Default: false
```

I don't see "more" infoirmation in the logs. 
I'm getting the same error message. Note that we get data for the `incoming messages`

```
[09:16:02 FTL] BadRequest: Failed to serialize response for request: https://westeurope.mdmrp.prod.windows.azure.net/subscriptions/6c53cca3-a4bc-49aa-b862-19f713f7b8eb/resourceGroups/eventhub-acc-rg/providers/Microsoft.EventHub/namespaces/stedin-dl-evh-acc/providers/microsoft.insights/metrics?$filter=EntityName eq '*'&timespan=2022-04-30T09%3A16%3A00.4876959Z%2F2022-05-05T09%3A16%3A00.4876959Z&interval=PT1M&metricnames=Size&aggregation=Average&top=10000&api-version=2018-01-01, Duration: 00:00:00.8906296, TraceId: {f64d3761-b289-4ff3-8e61-a186e8dde99f}
[09:16:03 INF] Found value 2132 for metric azure_event_hubs_incoming_messages with dimension raw.mfe-events.meterevent as part of entity_name dimension with aggregation interval 00:01:00
[09:16:03 INF] Found value 0 for metric azure_event_hubs_incoming_messages with dimension dnwg-his-scada.hlp-15mi-agg as part of entity_name dimension with aggregation interval 00:01:00
[09:16:03 INF] Found value 0 for metric azure_event_hubs_incoming_messages with dimension sql-his-scada.anli15mn as part of entity_name dimension with aggregation interval 00:01:00
[09:16:03 INF] Found value 0 for metric azure_event_hubs_incoming_messages with dimension sql-his-scada.anloth15mnkey as part of entity_name dimension with aggregation interval 00:01:00
[09:16:03 INF] Found value 0 for metric azure_event_hubs_incoming_messages with dimension sql-his-scada.anlu15mnkey as part of entity_name dimension with aggregation interval 00:01:00
[09:16:03 INF] Found value 0 for metric azure_event_hubs_incoming_messages with dimension sql-his-scada.anlu15mn as part of entity_name dimension with aggregation interval 00:01:00
[09:16:03 INF] Found value 0 for metric azure_event_hubs_incoming_messages with dimension sql-his-scada.anloth15mn as part of entity_name dimension with aggregation interval 00:01:00
```

**Comment by tomkerkhove** [comment]
Hm sorry to hear - I'l ltry to dive in to this asap.

**Comment by dks0296586** [comment]
I am also receiving BadRequest responses for EventHub, but they are slightly different errors

`   - name: azure_eventhubs_activeconnections_average
      description: "Total Active Connections for Microsoft.EventHub."
      resourceType: EventHubs
      azureMetricConfiguration:
        metricName: ActiveConnections
        aggregation:
          type: Average
      resourceDiscoveryGroups:
      - name: event-hubs-landscape
      
    - name: azure_eventhubs_activeconnections_maximum
      description: "Total Active Connections for Microsoft.EventHub."
      resourceType: EventHubs
      azureMetricConfiguration:
        metricName: ActiveConnections
        aggregation:
          type: Maximum
      resourceDiscoveryGroups:
      - name: event-hubs-landscape`

`[14:48:27 FTL] BadRequest: Metric: ActiveConnections does not support requested dimension combination: entityname, supported ones are: , TraceId: {xxxxx}
[14:48:29 FTL] BadRequest: Metric: ActiveConnections does not support requested dimension combination: entityname, supported ones are: , TraceId: {xxxxx}
[14:48:30 FTL] BadRequest: Metric: ActiveConnections does not support requested dimension combination: entityname, supported ones are: , TraceId: {xxxxx}
`

I noticed that the log message is indicating that the metric provided the dimension, entityname, however in my metric definition i do not. Also, microsoft docs(and log message) indicates there is no dimension supported for this metric.

**Comment by tomkerkhove** [comment]
Ha, but you are using resource discovery which will use that dimension under the hood I think. Let me check this as it might be a bug/known limitation.

**Comment by IMAMBAKS** [comment]
hi @tomkerkhove. Since this week we also have this problem with the incoming messages. 
We have **found THE problem** but we cannot solve it on our side. 

When we are looking at the URL, the timespan covers 5 days! but the interval is 1 minute.  We tested it with a smaller interval (1 minute). this worked fine for us.  See image below:

![image](https://user-images.githubusercontent.com/5062713/184350644-dc69ed76-f4b1-40cf-acc6-fe4f79fa91dc.png)

**Comment by tomkerkhove** [comment]
Interesting, seems like the requests are very greedy (by SDK or me). I'll check this ASAP

**Comment by tomkerkhove** [comment]
I started working on a fix: https://github.com/tomkerkhove/promitor/pull/2103

**Comment by tomkerkhove** [comment]
Re-opening for Helm changes

**Comment by IMAMBAKS** [comment]
@tomkerkhove, any idea when there will be an update to the helm chart? We still have this problem within our organisation unfortunately.

**Comment by tomkerkhove** [comment]
I'm a bit busy at the moment but it feels like the chart is having the problems from https://github.com/tomkerkhove/promitor/issues/2089 so I have to fix that. Sorry for the troubles.

**Comment by IMAMBAKS** [comment]
Is it possible to start scrape from "now", so that we can evade

**Comment by tomkerkhove** [comment]
What do you mean exactly?

**Comment by tomkerkhove** [comment]
This is available in Scraper v2.8.0 - https://github.com/tomkerkhove/promitor/releases/tag/Scraper-v2.8.0.

Helm chart will be updated shortly
